megalinter/flavor_factory.py

```diff
@@ -84,31 +84,34 @@ def check_active_linters_match_flavor(active_linters):
     missing_linters = []
     for active_linter in active_linters:
         if active_linter.name not in flavor_linters:
-            missing_linters += [active_linter.name]
             active_linter.is_active = False
-    # Manage cases where linters are missing in flavor
-    if len(missing_linters) > 0:
-        # Do not warn/stop if missing linters are repository ones (mostly OX.security related)
-        if not are_all_repository_linters(missing_linters):
```

Comment:
@Kurt-von-Laven plz could you rename are_all_repository_linters into are_ignorable_linters, and move your check inside the method ? thanks :)

Reply:
The reason I moved the check up here was that I didn't think we should be including the repository linters in `missing_linters` since we display its contents to the user. I can decompose the check (and rename the helper method) and retain the effect of the commit copied below, but it feels indirect to add the linters into this list only to then immediately remove them. I could alternatively revert the commit copied below (and rename the helper method) if you would rather we continue listing these linters as missing.

> Don't warn that REPOSITORY linters are missing
>
> Some REPOSITORY linters are intentionally omitted from flavors to keep their size small and performance fast. Hence, their absence from a flavor isn't considered an error when `FAIL_IF_MISSING_LINTER_IN_FLAVOR` is `true`. Therefore, omit them from the list of missing linters displayed to the user to avoid confusion.

Reply:
nah that's ok ^^

Reply:
@Kurt-von-Laven you broke something here :(
Now the beta version of python flavor crashes before it looks for devskim, which is not in this flavor on purpose :/

Reply:
Can you link me to or paste the stack trace or error message? All of the checks that I see passed when this PR was merged to main, so I'm guessing this is an issue that snuck past CI?

Reply:
https://github.com/oxsecurity/megalinter/actions/runs/4049785781/jobs/6966530825

Reply:
But the latest one works... I'm kind of lost, maybe it was just a version of beta that was wrongly built >_<
https://github.com/oxsecurity/megalinter/actions/runs/4052028152/jobs/6970984591

Reply:
Ah, yes, I remember seeing that error inconsistently while working on this pull request and I believe on #2275 as well, so I assumed it was unrelated to either. It sounds like this may be a job for [`git bisect`](https://git-scm.com/docs/git-bisect) since we aren't sure when it started, although it probably would have to be scripted to try several times since it's not reproducing consistently.

Reply:
It's back :(
https://github.com/oxsecurity/megalinter/actions/runs/4052519085/jobs/6972038673

Reply:
If I were to `cd mega-linter-runner` once and then `npm run test` repeatedly at each Git bisection point, would that be likely to isolate the issue to a particular commit?